SKYKitChat/Classes/UI/ParticipantList/SKYChatParticipantListViewController.swift

```diff
@@ -0,0 +1,307 @@
+//
+//  SKYChatParticipantListViewController.swift
+//  SKYKitChat
+//
+//  Copyright 2016 Oursky Ltd.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import SVProgressHUD
+
+public enum SKYChatParticipantQueryMethod: UInt {
+    // Initial value for the enum, for compatible with objective-c
+    case Undefined
+
+    // By username of the user, supposed to be unique
+    case ByUsername
+
+    // By email of the user, supposed to be unique
+    case ByEmail
+
+    // By name of the user, can limit the searching scope by setting participantScope
+    case ByName
+}
+
+open class SKYChatParticipantListViewController: UIViewController {
+
+    static let queryMethodCoderKey = "QUERY_METHOD"
+
+    public var skygear: SKYContainer?
+    public var queryMethod: SKYChatParticipantQueryMethod = .Undefined {
+        didSet {
+            self.searchBar.keyboardType = .default
+
+            switch self.queryMethod {
+            case .ByEmail:
+                self.searchBar.placeholder = "Search for user email"
+                self.searchBar.keyboardType = .emailAddress
+            case .ByUsername:
+                self.searchBar.placeholder = "Search for username"
+            case .ByName:
+                self.searchBar.placeholder = "Search for name of user"
+            default:
+                self.searchBar.placeholder = nil
+            }
+        }
+    }
+    public var participantScope: SKYQuery?
+    public var delegate: SKYChatParticipantListViewControllerDelegate?
+    internal(set) public var searchTerm: String?
+
+    @IBOutlet public var searchBar: UISearchBar!
+    @IBOutlet public var tableView: UITableView!
+
+    var participants: [SKYRecord] = []
+
+    // MARK: - Initializer
+
+    public init(queryMethod: SKYChatParticipantQueryMethod) {
+        super.init(nibName: "SKYChatParticipantListViewController",
+                   bundle: Bundle(for: SKYChatParticipantListViewController.self))
+        self.queryMethod = queryMethod
+    }
+
+    // MARK: - NSCoding Protocol
+
+    required public init?(coder aDecoder: NSCoder) {
```

Comment:
When user is trying to use our view controller classes in storyboard, the method `init?(coder aDecoder: NSCoder)` is called instead of `init(nibName:bundle:)`. You may check it on the example project.